kms/api-client/asymmetric.py

```diff
@@ -13,71 +13,100 @@
+    client = kms_v1.KeyManagementServiceClient()
+
+    # The resource name of the KeyRing associated with the CryptoKey.
+    parent = client.key_ring_path(project_id, location_id, key_ring_id)
+
+    # Create the CryptoKey object template
+    purpose = enums.CryptoKey.CryptoKeyPurpose.ASYMMETRIC_DECRYPT
+    algorithm = enums.CryptoKeyVersion.CryptoKeyVersionAlgorithm.\
+        RSA_DECRYPT_OAEP_2048_SHA256,
+    crypto_key = {'purpose': purpose,
+                  'version_template': {'algorithm': algorithm}}
+
+    # Create a CryptoKey for the given KeyRing.
+    response = client.create_crypto_key(parent, crypto_key_id, crypto_key)
+
+    print('Created CryptoKey {}.'.format(response.name))
+    return response
+# [END kms_create_asymmetric_key]
+
 
 # [START kms_get_asymmetric_public]
-def getAsymmetricPublicKey(client, key_path):
+def get_asymmetric_public_key(key_name):
     """
     Retrieves the public key from a saved asymmetric key pair on Cloud KMS
 
+    Example key_name:
+      "projects/PROJECT_ID/locations/global/keyRings/RING_ID/cryptoKeys/KEY_ID"
+
     Requires:
       cryptography.hazmat.backends.default_backend
       cryptography.hazmat.primitives.serialization
     """
-    request = client.projects() \
-                    .locations() \
-                    .keyRings() \
-                    .cryptoKeys() \
-                    .cryptoKeyVersions() \
-                    .getPublicKey(name=key_path)
-    response = request.execute()
-    key_txt = response['pem'].encode('ascii')
+
+    client = kms_v1.KeyManagementServiceClient()
+    response = client.get_public_key(key_name)
+
+    key_txt = response.pem.encode('ascii')
     key = serialization.load_pem_public_key(key_txt, default_backend())
     return key
 # [END kms_get_asymmetric_public]
 
 
 # [START kms_decrypt_rsa]
-def decryptRSA(ciphertext, client, key_path):
+def decrypt_rsa(ciphertext, key_name):
     """
     Decrypt the input ciphertext (bytes) using an
     'RSA_DECRYPT_OAEP_2048_SHA256' private key stored on Cloud KMS
 
-    Requires:
-      base64
+    Example key_name:
+      "projects/PROJECT_ID/locations/global/keyRings/RING_ID/cryptoKeys/KEY_ID"
     """
-    request_body = {'ciphertext': base64.b64encode(ciphertext).decode('utf-8')}
-    request = client.projects() \
-                    .locations() \
-                    .keyRings() \
-                    .cryptoKeys() \
-                    .cryptoKeyVersions() \
-                    .asymmetricDecrypt(name=key_path,
-                                       body=request_body)
-    response = request.execute()
-    plaintext = base64.b64decode(response['plaintext'])
-    return plaintext
+
+    client = kms_v1.KeyManagementServiceClient()
+    response = client.asymmetric_decrypt(key_name, ciphertext)
+    return response.plaintext
 # [END kms_decrypt_rsa]
 
 
 # [START kms_encrypt_rsa]
-def encryptRSA(plaintext, client, key_path):
+def encrypt_rsa(plaintext, key_name):
```

Comment:
Needs a CryptoKeyVersion name, not just a CryptoKey name, or else get_public_key will fail.